starlite/testing/async_test_client/transport.py

```diff
@@ -0,0 +1,187 @@
+        path = request.url.path
+        raw_path = request.url.raw_path
+        query = request.url.query.decode(encoding="ascii")
+        default_port = 433 if scheme in {"https", "wss"} else 80
+
+        if ":" in netloc:
+            host, port_string = netloc.split(":", 1)
+            port = int(port_string)
+        else:
+            host = netloc
+            port = default_port
+
+        host_header = request.headers.pop("host", host if port == default_port else f"{host}:{port}")
+
+        headers = [(k.lower().encode(), v.encode()) for k, v in (("host", host_header), *request.headers.items())]
+
+        return {
+            "type": "websocket" if scheme in {"ws", "wss"} else "http",
+            "path": unquote(path),
+            "raw_path": raw_path,
+            "root_path": self.root_path,
+            "scheme": scheme,
+            "query_string": query.encode(),
+            "headers": headers,
+            "client": ("testclient", 50000),
+            "server": (host, port),
+        }
+
+    async def handle_async_request(self, request: "Request") -> "Response":
+        scope = self.parse_request(request=request)
+        if scope["type"] == "websocket":
+            scope.update(
+                subprotocols=[value.strip() for value in request.headers.get("sec-websocket-protocol", "").split(",")]
+            )
+            session = WebSocketTestSession(client=self.client, scope=cast("WebSocketScope", scope))  # type: ignore
+            raise ConnectionUpgradeException(session)
+
+        scope.update(method=request.method, http_version="1.1", extensions={"http.response.template": {}})
+
+        raw_kwargs: Dict[str, Any] = {"stream": BytesIO()}
+
+        try:
+            with self.client.portal() as portal:
+                response_complete = portal.call(Event)
+                context: SendReceiveContext = {
+                    "response_complete": response_complete,
+                    "request_complete": False,
+                    "raw_kwargs": raw_kwargs,
+                    "response_started": False,
+                    "template": None,
+                    "context": None,
+                }
+                portal.call(
+                    self.client.app,
+                    scope,
+                    self.create_receive(request=request, context=context),
+                    self.create_send(request=request, context=context),
+                )
+        except BaseException as exc:  # pragma: no cover # pylint: disable=W0703
+            if self.raise_server_exceptions:
+                raise exc
+            return Response(
+                status_code=HTTP_500_INTERNAL_SERVER_ERROR, headers=[], stream=ByteStream(b""), request=request
+            )
+        else:
+            if not context["response_started"]:  # pragma: no cover
+                if self.raise_server_exceptions:  # noqa: SIM102
+                    if not context["response_started"]:
```

Comment:
this is the statement where flake8 has problems, even after changing the `assert statement` to `if statement` and running the flake8 pre-commit hook locally ( which passes ) it still has some issues here in the ci
````diff
- assert context["response_started"] , "TestClient did not receive any response."
+  if not context["response_started"]:
+ raise AssertionError("TestClient did not receive any response.")